guarddog/utils/archives.py

```diff
@@ -61,7 +46,7 @@ def safe_extract(source_archive: str, target_directory: str) -> None:
 
     """
     log.debug(f"Extracting archive {source_archive} to directory {target_directory}")
-    if is_tar_archive(source_archive):
+    if tarsafe.is_tarfile(source_archive):
```

Comment:
feels weird to interchange ours and the built-in function.
if we use `tarsafe.is_tarfile` in our function (`is_tar_archive`), we can just use ours everywhere.
same goes for `is_zip_archive`